petab/core.py

```diff
@@ -366,3 +366,17 @@ def _add_file_metadata(location: str, description: str = ""):
 
     archive.addMetadata(".", description)
     archive.writeToFile(filename)
+
+
+def unique_preserve_order(seq: Sequence) -> List:
+    """Return a list of unique elements in Sequence, keeping only the first
+    occurrence of each element
+
+    seq: Sequence to prune
+
+    Returns:
+        List of unique elements in ``seq``
+    """
+    seen = set()
+    seen_add = seen.add
+    return [x for x in seq if not (x in seen or seen_add(x))]
```

Comment:
funny but should work

Reply:
it does :sweat_smile: . was there before, just moved and documented.